src/solver/optim/ortoolsImpl/linearProblem.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include <exception>
+#include <memory>
+#include <ortools/linear_solver/linear_solver.h>
+
+#include <antares/solver/optim/ortoolsImpl/linearProblem.h>
+#include <antares/solver/utils/ortools_utils.h>
+
+namespace Antares::Solver::Optim::OrtoolsImpl
+{
+
+OrtoolsLinearProblem::OrtoolsLinearProblem(bool isMip, const std::string& solverName)
+{
+    auto* mpSolver = isMip ? MPSolver::CreateSolver(
+                               (OrtoolsUtils::solverMap.at(solverName)).MIPSolverName)
+                           : MPSolver::CreateSolver(
+                               (OrtoolsUtils::solverMap.at(solverName)).LPSolverName);
+
+    mpSolver_ = std::unique_ptr<operations_research::MPSolver>(mpSolver);
+    objective_ = mpSolver->MutableObjective();
+
+    params_.SetIntegerParam(MPSolverParameters::SCALING, 0);
+    params_.SetIntegerParam(MPSolverParameters::PRESOLVE, 0);
+}
+
+class ElemAlreadyExists: public std::exception
+{
+public:
+    const char* what() const noexcept override
+    {
+        return "Element name already exists in linear problem";
+    }
+};
+
+OrtoolsMipVariable* OrtoolsLinearProblem::addVariable(double lb,
+                                                      double ub,
+                                                      bool integer,
+                                                      const std::string& name)
+{
+    if (variables_.contains(name))
```

Comment:
if it turns out we cant avoid hosting **variables_** and **constraints_** : 
**Question** : 
Do the following instructions : 
- ` if (constraints_.contains(name)) `
- `if (!mapIteratorPair.second)`

check the same thing ?

If it turnes out they do, a more simplify form of this function : 
```C++
OrtoolsMipVariable* OrtoolsLinearProblem::addVariable(double lb,
                                                      double ub,
                                                      bool integer,
                                                      const std::string& name)
{
    if (variables_.contains(name))
    {
        logs.error() << "This variable already exists: " << name;
        throw ElemAlreadyExists();
    }

    auto* mpVar = mpSolver_->MakeVar(lb, ub, integer, name);

    if (!mpVar)
    {
        logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
    }

    const auto& pair = variables_.try_emplace(name, std::make_unique<OrtoolsMipVariable>(mpVar));
    return pair.first->second.get(); // <<name, var>, bool>
}
```

Reply:
Done in #2374